src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { Dispatch, SetStateAction, useRef, useState } from 'react';
+
+type Props = {
+  todo: Todo;
+  isLoading?: boolean;
+  isInEditMode?: boolean;
+  handleRemoveTodo: (todoId: number) => Promise<void>;
+  handleUpdateTodo: (todo: Todo) => Promise<void>;
+  setEditedTodoId: Dispatch<SetStateAction<null | number>>;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  isLoading,
+  handleRemoveTodo,
+  handleUpdateTodo,
+  isInEditMode,
+  setEditedTodoId,
+}) => {
+  const [todoTitleValue, setTodoTitleValue] = useState(todo.title);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const onCheckTodo = () => {
+    const todoToUpdate = { ...todo, completed: !todo.completed };
+
+    handleUpdateTodo(todoToUpdate);
+  };
+
+  const onDoubleClick = () => {
+    setEditedTodoId(todo.id);
+  };
+
+  const onBlur = async (
+    event: React.FormEvent<HTMLElement> | React.FocusEvent<HTMLFormElement>,
+  ) => {
+    event.preventDefault();
+
+    const normalizedTitle = todoTitleValue.trim();
+
+    if (todo.title === normalizedTitle) {
+      setEditedTodoId(null);
+
+      return;
+    }
+
+    if (normalizedTitle === '') {
```

Comment:
u can use this syntax for all the cases
```suggestion
    if (!normalizedTitle.length) {
```